crowbar_framework/spec/lib/crowbar/lock/shared_non_blocking_spec.rb

```diff
@@ -0,0 +1,199 @@
+require "spec_helper"
+require "timeout"
+require "rspec/expectations"
+
+RSpec::Matchers.define :be_unlocked do
+  match do |actual|
+    !actual.locked? && !File.exists?(actual.path)
+  end
+  failure_message do |actual|
+    if actual.locked?
+      "expected that #{actual} would not be locked"
+    else
+      "expected that #{actual.path} would not exist"
+    end
+  end
+end
+
+RSpec::Matchers.define :be_locked_by_me do
+  match do |actual|
+    actual.locked? &&
+      File.exists?(actual.path) &&
+      File.read(actual.path) =~ /acquired by #{actual.owner}$/
+  end
+  failure_message do |actual|
+    if !actual.locked?
+      return "expected that #{actual} would be locked"
+    end
+    if File.exists?(actual.path)
+      return "expected that #{actual.path} would exist"
+    end
+    return "expected #{actual.path} to match " +
+           "/acquired by #{actual.owner}$/:\n" +
+           File.read(actual.path)
```

Comment:
Style/MultilineOperationIndentation: Use 2 (not 7) spaces for indenting an expression spanning multiple lines.